src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,379 @@
+                for (final String citizenName : citizenNameStrings)
+                {
+                    final String[] citizenNameParts = citizenName.split(" ");
+                    if ((potentiaCitizenNameParts[0].equals(citizenNameParts[0])) && (potentiaCitizenNameParts[1].equals(citizenNameParts[1])))
+                    {
+                        lastNameSet.add(citizenNameParts[2]);
+                    }
+                }
+                return lastNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[2])).collect(Collectors.toList());
+           }
+        }
+        return Collections.emptyList();
+    }
+
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
+            @NotNull final List<ActionMenuHolder> parsedHolders,
+            final String potentialArgumentValue)
+    {
+        // TODO: selector support, such as used by CommandKill to find player
+        // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
+
+        switch (this)
+        {
+            case INTEGER:
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                return Ints.tryParse(potentialArgumentValue);
+            case BOOLEAN:
+                return Boolean.parseBoolean(potentialArgumentValue);
+            case PLAYER:
+                return parsePlayerValue(server, potentialArgumentValue);
+            case COLONY:
+                return parseColonyValue(sender, potentialArgumentValue);
+            case CITIZEN:
+                return parseCitizenDataValue(parsedHolders, potentialArgumentValue);
+            default:
+                throw new IllegalStateException("Unimplemented ActionArgumentType parsing");
+        }
+    }
+
+    private EntityPlayerMP parsePlayerValue(@NotNull final MinecraftServer server, final String potentialArgumentValue)
+    {
+        final List<String> playerNameStrings = getOnlinePlayerNames(server);
+        if (playerNameStrings.contains(potentialArgumentValue))
+        {
+            return server.getPlayerList().getPlayerByUsername(potentialArgumentValue);
+        }
+        else
+        {
+            if ("[abandoned]".equals(potentialArgumentValue))
+            {
+                return new FakePlayer(server.getWorld(0), new GameProfile(UUID.randomUUID(), "[abandoned]"));
+            }
+            return null;
+        }
+    }
+
+    private Colony parseColonyValue(@NotNull final ICommandSender sender, final String potentialArgumentValue)
+    {
+        final List<String> colonyNumberStrings = getColonyIdStrings();
+        final Integer result = Ints.tryParse(potentialArgumentValue);
+        if (null != result)
+        {
+            int colonyNumber = result.intValue();
+            if (sender instanceof EntityPlayer)
+            {
+                if (colonyNumber == -1)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)